engine/src/main/java/org/terasology/rendering/dag/nodes/BlurNode.java

```diff
@@ -0,0 +1,122 @@
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.dag.ConditionDependentNode;
+import org.terasology.rendering.dag.stateChanges.BindFBO;
+import org.terasology.rendering.dag.stateChanges.EnableMaterial;
+import org.terasology.rendering.dag.stateChanges.SetViewportToSizeOf;
+import org.terasology.rendering.opengl.BaseFBOsManager;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.opengl.FBOManagerSubscriber;
+
+import static org.terasology.rendering.opengl.OpenGLUtils.renderFullscreenQuad;
+
+/**
+ * A BlurNode takes the content of the color buffer attached to the input FBO and generates
+ * a blurred version of it in the color buffer attached to the output FBO.
+ */
+public class BlurNode extends ConditionDependentNode implements FBOManagerSubscriber {
+
+    private Material blurMaterial;
+    private float blurRadius;
+    private String performanceMonitorLabel;
+
+    private BaseFBOsManager fboManager;
+
+    private FBOConfig inputFBOConfig;
+    private FBOConfig outputFBOConfig;
+
+    private FBO inputFBO;
+    private FBO outputFBO;
+
+    /**
+     * Throws a RuntimeException if invoked. Use initialise(...) instead.
+     */
+    @Override
+    public void initialise() {
+        throw new RuntimeException("Please do not use initialise(). For this class use initialise(...) instead.");
+    }
+
+    /**
+     * Initializes the BlurNode instance.
+     *
+     * @param inputConfig an FBOConfig instance describing the input FBO, to be retrieved from the FBO manager
+     * @param outputConfig an FBOConfig instance describing the output FBO, to be retrieved from the FBO manager
+     * @param anfboManager the FBO manager from which to retrieve -both- FBOs.
+     * @param aRadius the blur radius: higher values cause higher blur. The shader's default is 16.0f.
+     * @param aLabel a String to label the instance's entry in output generated by the PerformanceMonitor
+     */
+    public void initialise(FBOConfig inputConfig, FBOConfig outputConfig, BaseFBOsManager anfboManager, float aRadius, String aLabel) {
+
+        this.inputFBOConfig = inputConfig;
+        this.outputFBOConfig = outputConfig;
+        this.fboManager = anfboManager;
+
+        requiresFBO(inputFBOConfig, fboManager);
+        requiresFBO(outputFBOConfig, fboManager);
+        addDesiredStateChange(new BindFBO(outputFBOConfig.getName(), fboManager));
+        addDesiredStateChange(new SetViewportToSizeOf(outputFBOConfig.getName(), fboManager));
+
+        update();
+        fboManager.subscribe(this);
+
+        setupConditions();
+
+        addDesiredStateChange(new EnableMaterial("engine:prog.blur"));
+        this.blurMaterial = getMaterial(new ResourceUrn("engine:prog.blur"));
+
+        this.blurRadius = aRadius;
+        this.performanceMonitorLabel = aLabel;
+    }
```

Comment:
This method will eventually become simpler and cleaner but I'll need to write some more StateChange subclasses for that, i.e. taking care of setting values in materials and binding input textures.